### PR TITLE
Add "Jetpack powered" badge to Notification Settings

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #NA

This PR adds a "Jetpack powered" badge below the "Your Sites" section, in Notification Settings.

Given that the changes only impacted `NotificationSettingsViewController`, and that the same had most of the code written between 2015 and 2016, I also took the occasion to modernize it a bit. There is still stuff that can be done, but did not want this PR to diverge.

iPad dark | iPhone light
-- | --
 | 

**KNOWN ISSUES: if you have many sites, the animation that expands and collapses the cells will generate a bit of funky movement of the badge. Since having many sites is an edge case, we decided that we can safely ship and then address the animation issue separately.**

To test:
- Build/run the `WordPress` target from this branch
- Go to Notifications then tap the cog in the top right to access Notification settings
- Make sure the "Jetpack powered" badge appears below the "Your Sites" section
- Stop the execution and build the `Jetpack` target
- Make sure you don't see the badge
- Stop the execution, change the `.jetpackPowered` Feature flag to false
- Build the `WordPress` target again and make sure the badge does not appear

## Regression Notes
1. Potential unintended areas of impact
`NotificationSettingsViewController`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test on a variety of devices and simulators

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] ~~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~~ Not yet released

### CODE REVIEW

**Review by guarani — COMMENTED** [review]
> - Stop the execution and build the Jetpack target
> - Make sure you don't see the badge

I'm seeing the badge on the Jetpack app as well, I ran the app from the PR's builds, not from Xcode, but I don't think that makes a difference.

| Screen Shot 2022-07-22 at 15 33 00 |
| - |

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingsViewController.swift [hunk 1]
```diff
@@ -342,8 +437,61 @@ open class NotificationSettingsViewController: UIViewController {
...
+                return NSLocalizedString("We’ll always send important emails regarding your account, " +
+                                         "but you can get some helpful extras, too.",
+                                         comment: "Title displayed in the Notification Settings for WordPress.com")
+            }
+        }
+
+        var showBadge: Bool {
+            switch self {
+            case .blog:
+                return true
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingsViewController.swift]: Is a check here needed for "feature flag on" && "app is WP"?

Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingsViewController.swift]: Thanks for catching this @Guarani !! It's indeed missing, will fix it momentarily!

**Comment by Gio2018** [comment]
Thanks for reviewing @Guarani ! It should be fixed now, ready for another pass!!

**Review by twstokes — APPROVED** [review]
LGTM @Gio2018. 🚀 

I tested with the WordPress app:
- Dark mode
- a11y sizes
- Different orientations

**Comment by guarani** [comment]
@Gio2018, I noticed that on Android, the Jetpack powered badge appears at the end of the screen (under the "Other" section). Here it appears under the "Your Sites" section. This doesn't seem like a blocker, but I was wondering which is correct.

**Comment by Gio2018** [comment]
> @Gio2018, I noticed that on Android, the Jetpack powered badge appears at the end of the screen (under the "Other" section). Here it appears under the "Your Sites" section. This doesn't seem like a blocker, but I was wondering which is correct.

Looking at the designs, it seems that it should be under the Your Sites section, would you agree @shaunandrews ?

**Review by guarani — APPROVED** [review]
I ran through the test cases and everything works 💯
